bot/extensions/adventofcode/commands.py

```diff
@@ -0,0 +1,222 @@
+    id: int
+    local_score: int
+
+    @validator("name", pre=True)
+    def set_name_to_anonymous(cls, val: Optional[str]) -> str:
+        if val is None:
+            return "Anonymous User"
+        return val
+
+
+class AdventOfCode(commands.GroupCog, group_name="aoc"):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @property
+    def role(self) -> discord.Role:
+        return self.bot.get_role(settings.aoc.role_id)
+
+    @app_commands.command()
+    async def subscribe(self, interaction: core.InteractionType):
+        """Subscribe to receive notifications for new puzzles"""
+
+        if self.role not in interaction.user.roles:
+            await interaction.user.add_roles(self.role)
+            await interaction.response.send_message(
+                "Okay! You have been __subscribed__ to notifications about new Advent of Code puzzles."
+                "You can run `/aoc unsubscribe` to disable them again for you.",
+                ephemeral=True,
+            )
+        else:
+            await interaction.response.send_message(
+                "Hey, you already are receiving notifications about new Advent of Code puzzles."
+                "If you don't want them any more, run `/aoc unsubscribe` instead.",
+                ephemeral=True,
+            )
+
+    @app_commands.command()
+    async def unsubscribe(self, interaction: core.InteractionType):
+        """Unsubscribe from receiving notifications for new puzzles"""
+
+        if self.role in interaction.user.roles:
+            await interaction.user.remove_roles(self.role)
+            await interaction.response.send_message(
+                "Okay! You have been __unsubscribed__ from notifications about new Advent of Code puzzles.",
+                ephemeral=True,
+            )
+        else:
+            await interaction.response.send_message(
+                "Hey, you don't even get any notifications about new Advent of Code puzzles currently anyway.",
+                ephemeral=True,
+            )
+
+    @app_commands.command()
+    async def countdown(self, interaction: core.InteractionType):
+        """Get the time left until the next puzzle is released"""
+
+        now = datetime.now(tz=pytz.timezone("EST"))
+        if now.month == 12:
+            # If it's December, calculate time until the next midnight EST time
+            target = now.replace(hour=0, minute=0, second=0, microsecond=0) + timedelta(days=1)
+        else:
+            # If it's not December, calculate time until the first of December
+            target = datetime(now.year, 12, 1, tzinfo=pytz.timezone("EST"))
+
+        if target < now:
+            return await interaction.response.send_message(
+                "Advent of Code is over for this year! See you next year!", ephemeral=True
+            )
+
+        await interaction.response.send_message(
+            f"There are {discord.utils.format_dt(target, 'R')} left until the next puzzle is released!",
```

Comment:
This would result in something like `"There are *in a year* left until the next puzzle is released!"`
 ```suggestion
            f"The next challenge is {discord.utils.format_dt(target, 'R')}",
```